### ISSUE TITLE
VSCode update mode settings do not work, auto-update cannot be disabled

### ISSUE THREAD

**Issue by ivaylo5ev** [microsoft/vscode#102666]
- VSCode Version: 1.46
- OS Version: Windows 10

Steps to Reproduce:
In relation to another issue (https://github.com/microsoft/vscode/issues/102118), I am trying to switch to an earlier version of VSCode.

1. Delete the `%APPDATA%\Code` directory (if having the latest version installed already, at the time of writing, version 1.47 is the latest version)
2. Remove the `C:\Program Files\Microsoft VS Code` folder.
3. Install an earlier version of VSCode (1.46.1 in my case).
4. Go to Settings and disable updates:
 - Set "Update Mode" to `none`.
 - Uncheck the "Enable Windows Background Update" setting.
5. VSCode prompts to restart. After a couple of restarts (and making sure the above update settings are set to prevent updates), VSCode would still auto-update. 

Does this issue occur when all extensions are disabled?: Yes

**Comment by ivaylo5ev** [comment]
For the scenario above I have used the System setup of VSCode 1.46.1

**Comment by joaomoreno** [comment]
Changing `update.mode` should be sufficient. Can you first do that and only then install the earlier version? I believe you still got an update because the update process started while you were on your way to change the setting.

**Comment by ivaylo5ev** [comment]
@joaomoreno  Installing an earlier version would require me to remove the previous installation. When I remove the previous installation, then the new install is with default settings. The update settings I listed are the first things I change and immediately restart the editor. After a couple of launches, or after a OS restart, the editor gets updated, so I believe it may not be related to the update process being started already.

If this helps -- I tried **user** setup for 1.44.2 (the issue is initially about the system-wide setup of vscode 1.46.1). With the user setup I repeat the same steps, but the update did not happen. Perhaps the problem is reproducible only for system-wide installs?

**Comment by joaomoreno** [comment]
>When I remove the previous installation, then the new install is with default settings.

ACtually settings are preserved, even if you uninstall. 👍

**Comment by ivaylo5ev** [comment]
Hello, @joaomoreno. Looks like you are correct, the updater seems to start on the first launch before I change update settings. I am performing a complete removal of Code, thus I am not keeping any settings. Nevertheless, I tried to install 1.46.1 (System) while I turned off the internet, so that the updater would fail. Then, I have changed `update.mode` to `none`, and re-enabled the internet connection, also restarted my OS a couple of times. All seems fine now I am staying with the 1.46.1 version.
